Fix material name conflicts

nanogram opened the pull request:
### Adds checkbox when importing wimdo & camdo.
Used mainly for importing many assets at once & needing unique materials.
This should solve the issue where multiple assets are imported & some materials share the same name, leading to meshes getting assigned incorrect materials.
Another option potentially is to use the armature or file name for material name suffix to differentiate.

nanogram committed:
Implement Material randomization

Used mainly for importing many assets at once & needing unique materials.

nanogram committed:
Remove bool & Update functions

- Removed randomize_name bool
- Removed extract_index
- Renamed extract_image_index to extract_index_name
- Updated instances of extract_index to extract_index_name

ScanMountGoat commented:
The `material_name` for wimdo and wismhd materials should start with `model_name` without the file extension similar to texture names. This should be good to merge after that change. Thank you for working on this.

nanogram committed:
Material Name Update

Materials now use the following naming: model_name.index.material_name.
Added "name" variable for extensionless filename, mirroring import_wismhd.

nanogram commented:
Should be all set now.

ScanMountGoat commented:
It looks like export_root.py is failing since it still uses the old function name. Importing looks good.

nanogram committed:
Updated export logic

Exports wimdo file now.  Some stuff just needed to be swapped around since previous extract_index did [int, str] while updated func has em as [str, int].

Throws error when reimporting generated file, but unsure if related to this pull since it occurs in v0.16.3. Maybe skill issue on my part.

ScanMountGoat commented:
Looks good. Thanks!
